controllers/openstackdataplanerole_controller.go

```diff
@@ -85,11 +86,78 @@ func (r *OpenStackDataPlaneRoleReconciler) SetupWithManager(mgr ctrl.Manager) er
 
 // ReconcileNodes ensure the desired state for the nodes
 func (r *OpenStackDataPlaneRoleReconciler) ReconcileNodes(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneRole) error {
-	// loop over r.Spec.DataPlaneNodes:
+	// loop over instance.Spec.DataPlaneNodes:
 	//   for each node:
 	//     (1) complete node.Spec based r.Spec.nodeTemplate, and values set on the
 	//         node (values on the node take precedence over those from the template)
 	//     (2) Create a CR of OpenStackDataPlaneNode from the node
 
+	for i := range instance.Spec.DataPlaneNodes {
+
+		// Setting OpenStackDataPlaneNodeName this way for now to workaround panic:
+		//   panic: runtime error: invalid memory address or nil pointer dereference
+		// OpenStackDataPlaneNodeName := instance.Spec.DataPlaneNodes[i].Node.HostName
+		OpenStackDataPlaneNodeName := fmt.Sprintf("dataplanenode-inheritance-%d", i)
+
+		node := &dataplanev1beta1.OpenStackDataPlaneNode{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      OpenStackDataPlaneNodeName,
+				Namespace: instance.Namespace,
+			},
+		}
+		// Set new node with values from its entry on the DataPlaneNodes list
+		node.Spec.Role = instance.Name
```

Comment:
Because the node is getting its role set here (line 109), I could changes lines 114 to 144 in this file to remove every `else` but keep every `if` so that each node is created with node properties only (besides the single reference to its role from line 109). Then on the node level, it can look up the value from the role and pass it to the Ansible call without setting it directly in the object itself. This assumes that we have the Ansible call from the node level but no Ansible call from the role level. I can try that in my next patch set.

Reply:
I've basically done this in the latest patch set. An Ansible call is not necessary to prove the inheritance works. Because each node automatically creates its own inventory in a configmap, you can observe the inventory configmap per node to see inheritance working.